django/db/backends/base/schema.py

```diff
@@ -31,10 +31,9 @@ def _is_relevant_relation(relation, altered_field):
 def _related_non_m2m_objects(old_field, new_field):
     # Filter out m2m objects from reverse relations.
     # Return (old_relation, new_relation) tuples.
-    return zip(
-        (obj for obj in old_field.model._meta.related_objects if _is_relevant_relation(obj, old_field)),
-        (obj for obj in new_field.model._meta.related_objects if _is_relevant_relation(obj, new_field))
-    )
+    for rel_name, old_rel in old_field.model._meta.fields_map.items():
+        if _is_relevant_relation(old_rel, old_field):
+            yield old_rel, new_field.model._meta.fields_map[rel_name]
```

Comment:
This looks unrelated.

Reply:
@felixxm without this in-place `test_alter_field_reloads_state_on_fk_with_to_field_related_name_target_type_change (migrations.test_operations.OperationTests)` fails on MariaDB

Reply:
Sounds good @felixxm